Celeste.Mod.mm/Patches/OuiChapterSelectIcon.cs

```diff
@@ -18,6 +57,87 @@ public patch_OuiChapterSelectIcon(int area, MTexture front, MTexture back)
             // no-op. MonoMod ignores this - we only need this to make the compiler shut up.
         }
 
+        public extern void orig_ctor(int area, MTexture front, MTexture back);
+        [MonoModConstructor]
+        public void ctor(int area, MTexture front, MTexture back) {
+            orig_ctor(area, front, back);
+
+            // Don't bother if its not in the path, perhaps replace with an alternative check for a .meta.yaml file?
+            if (!front.AtlasPath.Contains(DetectionPath))
+                return;
+
+            isAnimated = true;
+
+            // Remove the frame suffex.
+            string frontPath = Regex.Replace(front.AtlasPath, "\\d+$", string.Empty);
+            string backPath = frontPath + "_back";
+
+            FrontTextures = GFX.Gui.GetAtlasSubtextures(frontPath);
```

Comment:
So probably do the same thing here as with BackTextures and I think this code will work fine for un-animated icons as well.